packages/cli-repl/src/cli-repl.ts

```diff
@@ -795,6 +798,75 @@ export class CliRepl implements MongoshIOProvider {
     }
   }
 
+  verifyPlatformSupport(): void {
+    if (this.cliOptions.quiet) {
+      return;
+    }
+
+    // Typings for process.getReport haven't been updated
+    // (https://github.com/DefinitelyTyped/DefinitelyTyped/issues/40140)
+    const processReport = process.report?.getReport() as unknown as
+      | {
+          header: {
+            glibcVersionRuntime: string;
+          };
+        }
+      | undefined;
+    if (!processReport) {
+      return;
+    }
+
+    const satisfiesGLIBCRequirement = (glibcVersion: string): boolean => {
+      const RECOMMENDED_GLIBC_MAJOR = 2;
+      const RECOMMENDED_GLIBC_MINOR = 28;
+      // GLIBC versions don't necessarily have a patch version in them
+      // (https://sourceware.org/glibc/wiki/Glibc%20Timeline) which is why we
+      // don't exec for that
+      const GLIBC_REGEX = /(?<major>\d+)\.(?<minor>\d+)/;
+      const execResult = GLIBC_REGEX.exec(glibcVersion);
+      if (!execResult || !execResult.groups) {
+        return true;
+      }
+
+      const { major, minor } = execResult.groups;
+      if (parseInt(major, 10) < RECOMMENDED_GLIBC_MAJOR) {
+        return false;
+      }
+
+      if (parseInt(minor, 10) < RECOMMENDED_GLIBC_MINOR) {
+        return false;
+      }
+
+      return true;
+    };
+
+    const warnings: string[] = [];
+    if (!satisfiesGLIBCRequirement(processReport.header.glibcVersionRuntime)) {
+      warnings.push(
+        '  - Using mongosh on the current operating system is deprecated, and support may be removed in a future release.'
+      );
+    }
+
+    const RECOMMENDED_NODEJS = '>=20.0.0';
+    if (!semver.satisfies(process.version, RECOMMENDED_NODEJS)) {
+      warnings.push(
+        '  - Using mongosh with Node.js versions lower than 20.0.0 is deprecated, and support will be removed in a future release.'
+      );
+    }
+
+    if (warnings.length) {
+      const deprecationWarning = [
+        'Deprecation warnings:',
+        ...warnings,
+        'See https://www.mongodb.com/docs/mongodb-shell/ for documentation on supported platforms.',
```

Comment:
This I will confirm with @GaurabAryal wether it is ok to keep this or if already has a link for me to put here.

Reply:
Confirmed that we're leaving the link as is for now but I will create a follow up ticket for this link to be updated before releasing 2.x